pkg/distributor/distributor.go

```diff
@@ -556,6 +556,16 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 				continue
 			}
 
+			if ok, until := d.ShouldBlockPolicy(lbs, tenantID); ok {
```

Comment:
Also, we should pass the already resolved policy here to avoid resolving it twice.